studygroups/models.py

```diff
@@ -653,6 +657,36 @@ def get_top_discourse_topics_and_users(limit=10):
     top_posts_json = get_json_response("https://community.p2pu.org/top/monthly.json")
     return { 'topics': top_posts_json['topic_list']['topics'][:limit], 'users': top_posts_json['users'] }
 
+def get_active_teams():
+    today = datetime.datetime.now()
+    two_weeks_ago = today - relativedelta(days=+14)
+    memberships = StudyGroup.objects.published().filter(Q(start_date__gte=today) | Q(start_date__lte=today, end_date__gte=today) | Q(end_date__gt=two_weeks_ago, end_date__lte=today)).values_list('facilitator__teammembership', flat=True)
+    active_teams = Team.objects.filter(teammembership__in=memberships).distinct()
+    return active_teams
+
+def get_active_facilitators():
+    # it's not filtering out deleted studygroups
+    # it's not filtering users by the studygroup_count
+    # and it's not ordering correctly
+    # lajf;ajdf;jah;khak;gjha
+    facilitators = User.objects.annotate(\
```

Comment:
@dirkcuys can you help me with this query? I'm still not getting the right results, even though i'm following the Django 1.11 documentation and it's _kind of_ working... 
it seems like the `studygroup__deleted_at__isnull=True` condition is ignored, and `.filter(studygroup_count__gte=2).order_by('-studygroup_count')` is not working correctly, it's not doing nothing but it's inconsistent.

Reply:
@s-kennedy I think you're not getting the data that you're expecting. I didn't look at the SQL this query generates, but I'm suspecting that studygroup_count actually counts every application, not only the study_groups.

I started with 

```python 
User.objects.annotate(
    studygroup_count=Sum(Case(When(studygroup__draft=False, studygroup__deleted_at__isnull=True, then=1), default=Value(0), output_field=IntegerField()))
).order_by('-studygroup_count').values_list('username', 'studygroup_count')
```
and get the correct `studygroup_count`.

But when I add, 

```python
User.objects.annotate(
    studygroup_count=Sum(Case(When(studygroup__draft=False, studygroup__deleted_at__isnull=True, then=1), default=Value(0), output_field=IntegerField())),
    learners_count=Sum(Case(When(studygroup__draft=False, studygroup__deleted_at__isnull=True, studygroup__application__deleted_at__isnull=True, studygroup__application__accepted_at__isnull=False, then=1), output_field=IntegerField()))
).order_by('-studygroup_count').values_list('username', 'studygroup_count')
```

the totals goes way off.

Reply:
Okay, so this is the SQL query that gets you the data you want:

```sql
select 
   username, 
   max(lc.end_date),
   count(distinct(lc.id)) as lc_count,
   count(signup.email)
from
   auth_user as u,
   studygroups_studygroup as lc,
   studygroups_application as signup
where 
   lc.facilitator_id = u.id and signup.study_group_id = lc.id 
group by 
   u.username 
having 
   count(distinct(lc.id)) > 2 
order by 
   lc_count DESC;
```